components/DateInput.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { Controller, Control } from 'react-hook-form';
+import { LocalizationProvider, DatePicker } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+
+type DateInputProps = {
+  name: string;
+  control: Control;
+  label?: string;
+  rules?: object;
+  defaultValue?: string;
+};
+
+const DateInput = ({
+  name,
+  control,
+  label,
+  rules = {},
+  defaultValue = '',
+  ...rest
+}: DateInputProps) => (
+  <LocalizationProvider dateAdapter={AdapterDayjs}>
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      defaultValue={defaultValue}
+      render={({ field, fieldState }) => (
+        <DatePicker
+          label={label}
+          value={field.value || defaultValue}
```

Comment:
Incorrect logic: we should not be defaulting to a value of `defaultValue` if our `field.value` is `null`. If some date is set as default in the form for a given field, but I would like to remove it (leave it empty), I will never be able to do so.